crates/platform-node/src/hasher.rs

```diff
@@ -11,14 +10,9 @@ pub struct NodeTargetHasher {
     // Node.js version
     node_version: String,
 
-    // `package.json` `dependencies`
-    package_dependencies: BTreeMap<String, String>,
-
-    // `package.json` `devDependencies`
-    package_dev_dependencies: BTreeMap<String, String>,
-
-    // `package.json` `peerDependencies`
-    package_peer_dependencies: BTreeMap<String, String>,
+    // all the dependencies of the project (including dev and peer) and the hashes
+    // corresponding with their versions
+    dependencies: BTreeMap<String, Vec<String>>,
```

Comment:
Whys it a vec? Shouldnt the peer be resolved to whatever the dev dep is.

Reply:
In the example of https://github.com/moonrepo/moon/pull/403#issuecomment-1296592619, there are 3 versions of `execa` installed (through direct and indirect dependency tree) so they have 3 different integrity in `yarn.lock`. Better include all of them in the hashfile than using just one and having to figure out which one to use in subsequent runs.

Reply:
@IgnisDa We should sort the vec then so that the values are deterministic, otherwise it will produce different hashes.

Reply:
@milesj Yep makes sense. I pushed https://github.com/moonrepo/moon/pull/403/commits/9692c1b7abcc858cd98067633be5f58df5671989 that does it.